classes/class_06/class_06_homework.md

```diff
@@ -0,0 +1,163 @@
+<!-- $theme: default
+page_number: true
+footer: Java - Class 6 Homework
+-->
+# Home Work 
+
+# Note : Outputs expected to get after fixing errors if you notice any
+1). Consider the following 4 declarations:
+```
+   a). int i =10;
+   b). static int i=10;
+   c). public void m1()  { System.out.println(i); }
+   d). public static void m1() { System.out.println(i); }
+```   
+  Within the same class which of the following declarations are valid.
+    1). a & c
+	2). a & d
+	3). b & c
+	4). b & d
+
+ -----------------------------------------------------------------------------
+2). What will be the output of below code: 
+```
+  class Sample {
+   int i=10;
+   static int j=20;
+    
+   public static void main(String args[]) {
+     System.out.println(i+"...."+j); 
+         Sample s1=new Sample();
+         s1.i=100;
+         s1.j=200;
+     System.out.println(s1.i+"...."+s1.j); 	  
+	 Sample s2=new Sample();
+     System.out.println(s2.i+"...."+s2.j); 	  
+         s2.j=2000;
+     System.out.println(s1.i+"...."+s1.j); 
+     
+     }	  
+  }
+  ```
+-----------------------------------------------------------------------------
+
+ 3). Which of the following classes are tightly encapsulated ?
+```
+   class A { private int x=10; }
+   class B extends A { int y=20; }
+   
+   class C { int a=10; }
+   class D extends C { private int b=10; }
+   class E extends D { private int c=5; } 
+  ```
+ 
+-----------------------------------------------------------------------------
+ 4). What will be the output of below code:
+ ```
+   class Test {
+   int x;
+	 
+   Test() {
+   this.x=10;
+   super();	}
```

Comment:
We haven't gone over inheritance, so this will be confusing.

Reply:
to the extent needed to understand the concept, i will explain them in the class.